libnmstate/ifaces/ovs.py

```diff
@@ -184,6 +186,29 @@ def parent(self, value):
     def need_parent(self):
         return True
 
+    @property
+    def is_patch_port(self):
+        if self._config:
+            return True
+        else:
+            return False
+
+    def pre_edit_validation_and_cleanup(self):
+        super().pre_edit_validation_and_cleanup()
+        self._validate_ovs_interface_patch()
+
+    def _validate_ovs_interface_patch(self):
+        if self.is_patch_port and (
+            super().is_ipv4_enabled()
+            or super().is_ipv6_enabled()
+            or super().mtu
+            or super().mac
+        ):
```

Comment:
IMO, yes we should fail in that case. The user should disable the ip and remove the mtu and mac from the desired state if they want to convert an internal port to patch port.

What do you think? If we allow users to include mtu, mac or ip enabled the desired state it will be an inconsistent state..

Reply:
Those info will be merged from current state, the you fail on this merged state.

Reply:
Done, thanks!